frontend/app/src/components/settings/FrontendSettings.vue

```diff
@@ -3,22 +3,39 @@
     <template #title>
       {{ $t('frontend_settings.title') }}
     </template>
-    <v-switch
-      :value="!animationsEnabled"
-      class="general-settings__fields__animation-enabled mt-0"
-      :label="$t('frontend_settings.label.animations')"
-      :success-messages="settingsMessages[ANIMATIONS_ENABLED].success"
-      :error-messages="settingsMessages[ANIMATIONS_ENABLED].error"
-      @change="onAnimationsEnabledChange($event)"
-    />
-    <v-switch
-      v-model="scrambleData"
-      class="general-settings__fields__scramble-data"
-      :label="$t('frontend_settings.label.scramble')"
-      :success-messages="settingsMessages[SCRAMBLE_DATA].success"
-      :error-messages="settingsMessages[SCRAMBLE_DATA].error"
-      @change="onScrambleDataChange($event)"
-    />
+
+    <settings-option
+      #default="{ error, success, update }"
+      setting="animationsEnabled"
+      :session-setting="true"
+      :transform="value => !value"
+      :error-message="$tc('frontend_settings.validation.animations.error')"
+    >
+      <v-switch
+        :value="!animationsEnabled"
+        class="general-settings__fields__animation-enabled mt-0"
+        :label="$t('frontend_settings.label.animations')"
+        :success-messages="success"
+        :error-messages="error"
+        @change="update"
+      />
+    </settings-option>
+
+    <settings-option
+      #default="{ error, success, update }"
+      setting="scrambleData"
+      :session-setting="true"
```

Comment:
Since this is a boolean property we can use it as a flag so we should be able to do `session-setting` instead of `:session-setting="true"` can you change the instances of this and `:frontend-setting="true"` to the flag format? 


I found this rule https://eslint.vuejs.org/rules/prefer-true-attribute-shorthand.html so maybe we can activate it.